internal/index/note_index.go

```
@@ -0,0 +1,339 @@
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		if !strings.HasSuffix(d.Name(), ".md") {
+			return nil
+		}
+
+		rel, err := filepath.Rel(i.root, path)
+		if err != nil {
+			return nil
```

golangci-lint (2.11.4)

[error] error is not nil (line 94) but it returns nil

(nilerr)